code/lib/preview/src/utils.ts

```diff
@@ -0,0 +1,29 @@
+import { global } from '@storybook/global';
+import type { BrowserInfo } from 'browser-dtector';
+import BrowserDetector from 'browser-dtector';
+
+let browserInfo: BrowserInfo | undefined;
+
+function getBrowserInfo() {
+  if (!browserInfo) {
+    browserInfo = new BrowserDetector(global.navigator?.userAgent).getBrowserInfo();
+  }
+
+  return browserInfo;
+}
+
+export function preprocessError(
```

Comment:
I think this needs a code comment explaining what it's for.
Or perhaps it should be renamed to `prepareForTelemetry`?